modules/SemanticWebImport/src/main/java/fr/inria/edelweiss/semantic/importer/SemanticWebImportParser.java

```diff
@@ -0,0 +1,170 @@
+import org.gephi.graph.api.Table;
+
+
+/**
+ * Responsible for making the request and building the resulting graph
+ * asynchronously. The purpose of this class is to: <ol> <li>use a SPARQL driver
+ * to obtain a RDF/XML result;</li> <li>populate the graph by extracting in the
+ * result the nodes and the edges.</li> </ol>
+ *
+ * @author Erwan Demairy <Erwan.Demairy@inria.fr>
+ */
+public class SemanticWebImportParser implements LongTaskListener {
+
+	private static final Logger logger = Logger.getLogger(SemanticWebImportParser.class.getName());
+	private ProjectController pc;
+	private Graph rdfGraph;
+	private RequestParameters parameters;
+	private RdfAnalyzer analyzer;
+	private SparqlDriver driver;
+	private Semaphore waitEndPopulate = new Semaphore(0);
+	private LongTaskListener listener; // listener waiting for the end of the population task.
+	LongTaskExecutor executor = new LongTaskExecutor(true);
+
+	static public class RequestParameters {
+
+		private String rdfRequest;
+
+		public RequestParameters(String rdfRequest) {
+			setRdfRequest(rdfRequest);
+		}
+
+		public String getRdfRequest() {
+			return this.rdfRequest;
+		}
+
+		public final void setRdfRequest(String nodeRdfRequest) {
+			this.rdfRequest = nodeRdfRequest;
+		}
+	}
+
+	public SemanticWebImportParser() {
+		this(new RequestParameters(""));
+	}
+
+	public SemanticWebImportParser(RequestParameters requests) {
+		this.parameters = requests;
+	}
+
+	/**
+	 * Asynchronous operation that fills the current workspace.
+	 *
+	 * @sa waitEndpopulateRDFGraph
+	 *
+	 * @param driverUsed SPARQL request driver.
+	 * @param properties Parameters to use.
+	 */
+	public final void populateRDFGraph(SparqlDriver driverUsed, Properties properties, LongTaskListener listener) {
+		this.listener = listener;
+		boolean resetWorkspace = Boolean.parseBoolean(properties.getProperty(PluginProperties.RESET_WORKSPACE.getValue(), "false"));
+		boolean postProcessing = Boolean.parseBoolean(properties.getProperty(PluginProperties.POST_PROCESSING.getValue(), "false"));
+		String saveResultName = properties.getProperty(PluginProperties.SAVE_SPARQL_RESULT.getValue(), "");
+		int fynLevel = Integer.parseInt(properties.getProperty(PluginProperties.FYN_LEVEL.getValue(), "0"));
+
+		logger.log(Level.INFO, "resetWorkspace = {0}", resetWorkspace);
+		logger.log(Level.INFO, "postProcessing = {0}", postProcessing);
+
+		this.driver = driverUsed;
+		this.driver.setPluginProperties(properties);
+
+		GephiUtils.createProjectIfEmpty();
+		GephiUtils.createWorkspaceIfEmpty();
+
+		pc = Lookup.getDefault().lookup(ProjectController.class);
+		Workspace dataWorkspace = pc.getCurrentWorkspace();
+		GraphModel model = getCurrentGraphModel(dataWorkspace);
+		// @TODO how to reset the graph
+//        if (resetWorkspace) {
+//            model.getNodeTable().clear();
+//        }
+		AttributeColumnsController ac = Lookup.getDefault().lookup(AttributeColumnsController.class);
```

Comment:
This is unused. You can remove the 3 parts in the code where AttributeColumnsController is unsed and then remove datalab-api dependency altogether.